neurom/apps/morph_stats.py

```diff
@@ -164,17 +166,17 @@ def extract_stats(neurons, config):
     {config_path}
     """
 
-    def _fill_stats_dict(data, stat_name, stat):
+    def _fill_stats_dict(data, stat_name, stat, feature_name):
         """Insert the stat data in the dict.
 
         And if the stats is a 3D array, splits it into XYZ components.
         """
-        if stat is None or not isinstance(stat, np.ndarray) or stat.shape not in ((3, ), ):
-            data[stat_name] = stat
-        else:
+        if _find_feature_func(feature_name).shape[-1] == 3:
```

Comment:
Minor, what about comparing shape with `(..., 3)` instead of `shape[-1] == 3`? They should give the same result with the features defined at the moment, but it looks more explicit to me (and in general shape can be a tuple with 0 or more elements).